src/main/java/com/aws/greengrass/mqtt/bridge/clients/LocalMqtt5Client.java

```diff
@@ -589,31 +602,34 @@ private boolean retryUnsubscribe(UnsubAckPacket.UnsubAckReasonCode rc) {
 
     @Override
     public void start()  {
-        if (clientStarted) {
-            return;
+        if (isSSL()) {
+            mqttClientKeyStore.listenToUpdates(onKeyStoreUpdate);
         }
         client.start();
-        clientStarted = true;
     }
 
     @Override
     public void stop() {
-        mqttClientKeyStore.unsubscribeFromCAUpdates(onKeyStoreUpdate);
-        cancelUpdateSubscriptionsTask();
+        mqttClientKeyStore.unsubscribeFromUpdates(onKeyStoreUpdate);
+        cancelResetRetry();
         closeClient();
     }
 
     private void closeClient() {
+        cancelUpdateSubscriptionsTask();
+        doCloseClient();
+    }
+
+    private void doCloseClient() {
         try {
-            if (clientStarted) {
+            if (client.getIsConnected()) {
                 client.stop(null);
             } else {
                 client.close();
             }
         } catch (CrtRuntimeException e) {
-            LOGGER.atError().setCause(e).log("Failed to stop MQTT5 client");
-        } finally {
-            clientStarted = false;
+            LOGGER.atDebug().setCause(e).log("Failed to stop MQTT5 client");
```

Comment:
why is this debug? This sounds like a problem

Reply:
fixed